### PR TITLE
Test: integration tests use latest TestServer.apk

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

We should run the integration tests against the latest TestServer.apk + AndroidManifest.xml.

The integration tests skip the blocking version check in calabash-android.

The run_and_compile.sh script looks for local calabash-android sources using `CALABASH_ANDROID_PATH` or a relative directory (most common location for gem sources).  If no local sources are found, a git clone is performed to fetch sources from /master.

The integration tests Gemfiles have been updated to point to local calabash-android sources.

The latest TestServer.apk and AndroidManifest.xml are added to the gem sources.

### CODE REVIEW

**Comment by jmoody** [comment]
Related to https://github.com/calabash/calabash-android/pull/881

File: server/integration-tests/run_and_compile.sh [hunk 1]
```diff
@@ -2,15 +2,67 @@
 
 set -e
 
+source log.sh
+
 # Switch Java to v8 in runtime from current context
 . switch_java_version.sh
 
+GEM_DIR=calabash-android
```
Comment (AndreyMitsyk) [inline server/integration-tests/run_and_compile.sh]: Could you please move new logic from here to separate .sh file? This file is more like common file that executes required steps. If we will have separate file, we can add it to `fast.sh`

**Comment by jmoody** [comment]
Good point.

**Comment by jmoody** [comment]
@AndreyMitsyk Please don't merge yet.  I want to pair to discuss some documentation, do some knowledge transfer (from you to me), and talk about a way to organize the build scripts.

**Comment by jmoody** [comment]
@AndreyMitsyk Can you take another look at these changes?

Please make any changes you want and then merge.

**Pull request merged**